src/Model/LDAPGateway.php

```diff
@@ -57,7 +80,17 @@ public function __construct()
     protected function search($filter, $baseDn = null, $scope = Ldap::SEARCH_SCOPE_SUB, $attributes = [], $sort = '')
     {
         $records = $this->ldap->search($filter, $baseDn, $scope, $attributes, $sort);
+        return $this->processSearchResults($records);
+    }
 
+    /**
+     * Processes results from either self::search() or self::searchAll(), expecting eitheran array of records
+     *
+     * @param Iterator $records
+     * @return array
+     */
+    private function processSearchResults($records)
```

Comment:
Could we use protected instead of private wherever possible? The iterator class is probably fine because its marked as final already

Reply:
Yep no problem, that makes sense.

Reply:
Will leave you to mark resolved when you're satisfied, sorry!

Reply:
Still private